frontend/controller/actions/chatroom.js

```diff
@@ -3,13 +3,52 @@ import sbp from '@sbp/sbp'
 
 import { GIErrorUIRuntimeError, L } from '@common/common.js'
 import { has, omit } from '@model/contracts/shared/giLodash.js'
+import { CHATROOM_TYPES, MESSAGE_RECEIVE_RAW, MESSAGE_TYPES } from '@model/contracts/shared/constants.js'
 import { GIMessage } from '~/shared/domains/chelonia/GIMessage.js'
 import { Secret } from '~/shared/domains/chelonia/Secret.js'
 import { encryptedOutgoingData, encryptedOutgoingDataWithRawKey } from '~/shared/domains/chelonia/encryptedData.js'
 // Using relative path to crypto.js instead of ~-path to workaround some esbuild bug
 import { CURVE25519XSALSA20POLY1305, EDWARDS25519SHA512BATCH, deserializeKey, keyId, keygen, serializeKey } from '../../../shared/domains/chelonia/crypto.js'
 import type { GIRegParams } from './types.js'
 import { encryptedAction, encryptedNotification } from './utils.js'
+import { makeMentionFromUserID } from '@model/chatroom/utils.js'
+import messageReceivePostEffect from '@model/notifications/messageReceivePostEffect.js'
+
+sbp('okTurtles.events/on', MESSAGE_RECEIVE_RAW, ({
+  contractID,
+  data,
+  innerSigningContractID,
+  newMessage
+}) => {
+  const getters = sbp('state/vuex/getters')
+  const mentions = makeMentionFromUserID(getters.ourIdentityContractId)
+  const msgData = newMessage || data
+  const isMentionedMe = (!!newMessage || data.type === MESSAGE_TYPES.TEXT) && msgData.text &&
+  (msgData.text.includes(mentions.me) || msgData.text.includes(mentions.all))
+
+  if (!newMessage) {
+    const isAlreadyAdded = !!getters
+      .chatRoomUnreadMessages(contractID).find(m => m.messageHash === data.hash)
+
+    if (!isMentionedMe) {
+      sbp('gi.actions/identity/kv/removeChatRoomUnreadMessage', { contractID, messageHash: data.hash })
+    }
```

Comment:
Before, in `editMessage`, this would get called only when `isAlreadyAdded` were true... now the logic seems to have slightly changed.

Is this intentional? (if so, why?)

Reply:
You're right. It shouldn't make much of a difference, but I can amend that. I tried to simplify the logic and maybe went too far.